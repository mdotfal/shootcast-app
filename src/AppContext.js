import React from 'react';

const AppContext = React.createContext({
  lists: [],
  cities: [],
  onAddCity: () => {},
  onDeleteCity: () => {},
  onAddList: () => {},
  onDeleteList: () => {},
  onSignOut: () => {},
})

export default AppContext;