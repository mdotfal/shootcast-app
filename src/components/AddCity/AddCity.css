
@import url('https://fonts.googleapis.com/css2?family=Noto+Sans+JP:wght@400;500&display=swap');

/*  AddCity Styles */

.addcity-form {
  font-family: 'Noto Sans JP', sans-serif;
  position: relative;
  display: flex;
  align-items: center;
  flex-direction: column;
  padding: 10px;
  margin-bottom: 0;
}

input {
  margin: 2px;
}

select {
  border: none;
  border-bottom: 1px solid black;
}

fieldset {
  border-radius: 20px;
}