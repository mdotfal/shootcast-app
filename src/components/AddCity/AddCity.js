import React, { Component } from 'react';
import PropTypes from 'prop-types';
import AppContext from '../../AppContext';
import config from '../../config';
import './AddCity.css';

class AddCity extends Component {

  static contextType = AppContext;

  onSubmit = e => {
    e.preventDefault();
    const newCity = {
      name: e.target.search.value,
      list_id: e.target.listId.value
    }
    fetch( `${ config.API_ENDPOINT }/api/cities`, {
      method: 'POST',
      headers: {
        'content-type': 'application/json'
      },
      body: JSON.stringify( newCity )
    })
    .then( res => res.json())
    .then( data => {
      this.context.onAddCity( data );
      e.target.search.value = '';
    })
    .catch( error => { console.log( error )})
  }

  render() {
    return (
      <div className="addcity">
        <form className="addcity-form" onSubmit={ this.onSubmit }>
          <label htmlFor="search">
            <p>Add a City to your List</p>
            
          </label>
          <input type="text" placeholder="Enter City" name="search" id="search" required />
          <label htmlFor="listId">
            <p>Select a list.</p>
          </label>
          <select id="listId" name="listId">
            { this.context.lists.map( ( list, i ) => 
              <option
                key={ i }
                value={ list.id }
              >{ list.name }</option> 
            )}
          </select>
            <button
              type="submit"
            >Add City</button>
          </form>
      </div>
    );
  };
};

AddCity.propTypes = {
  lists: PropTypes.array,
}


export default AddCity;