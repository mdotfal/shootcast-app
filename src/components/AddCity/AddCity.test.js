import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import AddCity from './AddCity';
import AppContext from '../../AppContext';

describe.skip( `AddCity component`, () => {

  it( `renders without crashing`, () => {
    const div = document.createElement( 'div' );
    ReactDOM.render(
      <AppContext.Provider>
        <AddCity />
      </AppContext.Provider> 
    , div )
    ReactDOM.unmountComponentAtNode( div );
  })

  it(Â `renders without errors`, () => {
    const wrapper = shallow( 
      <AppContext.Provider>
        <AddCity /> 
      </AppContext.Provider>
    )
    expect( toJson( wrapper )).toMatchSnapshot()
  })
})