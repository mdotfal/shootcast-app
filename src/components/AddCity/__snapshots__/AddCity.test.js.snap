// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCity component renders without errors 1`] = `
<AddCity
  context={
    Object {
      "lists": Array [],
    }
  }
/>
`;
